main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -574,8 +574,8 @@ object Lowering {
         val parExp = mkParApply(Expression.Apply(e, es, t, pur, eff, loc1))
         Expression.Cast(parExp, None, Some(Type.Pure), Some(Type.Empty), t, pur, eff, loc0)
 
-      case _ =>
-        throw InternalCompilerException("Unexpected par expression")
+      case e =>
```

Comment:
No, I mean, you should write it out `case Exp.Par => `.

We want it so that when we add an new expression `Foo` then we get "Unmatched Foo`-- which the wildcard is blocking :)

Reply:
I'm not sure what you mean. Do you mean that we should add a case for each expression within in the par case (we could add a `visitParExp` helper function for this of course)?

Reply:
I.e. an exhaustive match

Reply:
You should have an explicit case for Par (like all the other cases).

Reply:
Oh so you want to handle the case `par par f(x)`?

Reply:
No, thats illegal.

Reply:
Okay, I get it now. I think maybe I was wrong.